elastic_service/elastic_handler.go

```diff
@@ -102,39 +103,39 @@ func buildSearchQuery(client *elastic.Client, title string, author string, price
 	builder := client.Search().Index(consts.Index).Query(all).Pretty(true)
 	q := elastic.NewBoolQuery()
 
-	if title != "" {
-		q.Must(elastic.NewMatchQuery(consts.Title, title))
+	if searchParams["title"] != "" {
+		q.Must(elastic.NewMatchQuery(elastic_fields.Title, searchParams["title"]))
 	}
-	if author != "" {
-		q.Must(elastic.NewMatchQuery(consts.Author, author))
+	if searchParams["author"] != "" {
+		q.Must(elastic.NewMatchQuery(elastic_fields.Author, searchParams["author"]))
 	}
 	if to != 0 {
-		q.Must(elastic.NewRangeQuery(consts.Price).From(from).To(to))
+		q.Must(elastic.NewRangeQuery(elastic_fields.Price).From(from).To(to))
 	}
 	builder = builder.Query(q)
 	return builder, nil
 }
 
-func (e *ElasticHandler) Search(title string, author string, priceRange string) (res []models.Book, err error) {
-
-	builder, err := buildSearchQuery(e.Client, title, author, priceRange)
+func (e *ElasticHandler) Search(searchParams map[string]string) (res []models.Book, err error) {
+	builder, err := buildSearchQuery(e.Client, searchParams)
 	if err != nil {
 		return nil, err
 	}
 	searchResult, err := builder.Pretty(true).Size(config.MaxQueryResults).Do(context.TODO())
-
 	if err != nil {
-		return nil, errors.Wrap(err, fmt.Sprintf("search failed for title: %s, author: %s, priceRange: %s", title, author, priceRange))
+		return nil, errors.Wrap(err, fmt.Sprintf("search failed for title: %s, author: %s, priceRange: %s", searchParams["title"], searchParams["author"], searchParams["priceRange"]))
```

Comment:
You can just return the `searchParams` map without breaking it down

Reply:
I'm printing the values, not returning it. I didn't find a pretty way to print all the map without extracting its fields

Reply:
WDYT about something like [this](https://go.dev/play/)

Reply:
It looks like they also broke down the map into it's fields, but I'll use it anyway since it's more generic